iOS Client/PodcastsController/PodcastDetailViewController.swift

```diff
@@ -41,19 +53,43 @@ class PodcastDetailViewController: UIViewController, StoreSubscriber {
         MBStore.sharedStore.dispatch(PopCurrentNavigation())
     }
 
+    @IBAction func pressPlayPause(_ sender: Any) {
+        MBStore.sharedStore.dispatch(PlayPausePodcast())
+    }
+    
+    @objc func onSeek(slider: UISlider, event: UIEvent) {
+        if let touchEvent = event.allTouches?.first {
+            let secondToSeekTo = Double(slider.value) * (totalDuration ?? 0.0)
+            switch touchEvent.phase {
+            case .moved:
+                updateCurrentDuration(current: secondToSeekTo, total: totalDuration ?? 0.0)
+            case .ended:
+                delegate?.seek(to: secondToSeekTo)
+            default:
+                break
+            }
+        }
+    }
+    
     static func instantiateFromStoryboard() -> PodcastDetailViewController {
         // swiftlint:disable force_cast
         return UIStoryboard(name: "Main", bundle: nil).instantiateViewController(withIdentifier: "PodcastDetailViewController") as! PodcastDetailViewController
         // swiftlint:enable force_cast
     }
     
     func newState(state: MBAppState) {
-        podcast = state.podcastsState.selectedPodcast
-        titleLabel.text = podcast?.title
-        if let guid = podcast?.guid, let url = URL(string: guid) {
-            let item = AVPlayerItem(url: url)
-            player = AVPlayer(playerItem: item)
-            player?.play()
+        switch state.podcastsState.player {
+        case .error, .initialized, .paused, .finished:
+            playPauseButton.setImage(#imageLiteral(resourceName: "play"), for: .normal)
+        case .playing:
+            playPauseButton.setImage(#imageLiteral(resourceName: "pause"), for: .normal)
         }
+        titleLabel.text = state.podcastsState.selectedPodcast?.title
+    }
+    
+    func updateCurrentDuration(current: Double, total: Double ) {
+        totalDuration = total
+        durationLabel.text = "\(formatter.string(from: NSNumber(value: current)) ?? "0") / \(formatter.string(from: NSNumber(value: total)) ?? "0")"
+        durationSlider.setValue(Float(current/(totalDuration ?? 0.0)), animated: true)
```

Comment:
divide by 0

Reply:
just use `total` instead of `totalDuration` here, and maybe do a guard to make sure total is not equal to 0 before doing this line